modules/shapefile/src/shapefile-loader.js

```diff
@@ -0,0 +1,59 @@
+import {SHPLoader} from './shp-loader';
+/** @typedef {import('@loaders.gl/loader-utils').LoaderObject} LoaderObject */
+
+// __VERSION__ is injected by babel-plugin-version-inline
+// @ts-ignore TS2304: Cannot find name '__VERSION__'.
+const VERSION = typeof __VERSION__ !== 'undefined' ? __VERSION__ : 'latest';
+
+/** @type {LoaderObject} */
+export const ShapeLoader = {
+  id: 'shp',
+  name: 'Shapefile',
+  category: 'geometry',
+  version: VERSION,
+  extensions: ['shp'],
+  mimeTypes: [],
+  options: {
+    shapefile: {}
+  },
+  parse: parseShapefile
+};
+
+async function parseShapefile(arrayBuffer, options, context) {
+  const {parse, fetch, baseName} = context;
+
+  // Attempt a parallel load of the small sidecar files
+  // TODO: Can these extensions be both lower and uppercase? Match the main extension?
```

Comment:
I think according to the spec, extensions _should_ be lower case, but that doesn't mean they always are. See:

https://github.com/calvinmetcalf/shapefile-js/issues/64
https://github.com/mapserver/mapserver/issues/4712